### PR TITLE
feat: read avatar url from gitlabs project details

### LINKED ISSUE #1242: Add project image
As a graph API user, I'd like the Cross-Entity Search and Single Project APIs to return info about Project image.

The problem & solution
Our UI team would like to display the Project image for the user. Currently, only Datasets have images on KG. Before Project image(s) will be introduced on the CLI, we can use Project image that can be retrieved from the GL API.

Acceptance criteria:

- [x] adapt the transformation process
  - [x] read `avatar_url` returned by the [Single Project GL API](https://docs.gitlab.com/ee/api/projects.html#get-single-project) during event transformation
  - [x] encode the URL as an image URI similar to Datasets; we should extract the Image model from Dataset to some common space and share it with Project
  - [x] ensure the URL can be changed so the transformation process needs to update/delete it in the TS
- [x] adapt the Cross-Entity Search to return info about Project image and a direct link to it (see the Dataset images)
  - [x] update README
- [x] adapt the Single Project API to return info about Project image and a direct link to it (see the Dataset images)
  - [x] update README

Assumption:
* we won't add images to projects that are already in the TS (no re-provisioning)

### PR DESCRIPTION
**Pull request by eikek**
Read `avatar_url` from gitlabs single-project response.

Issue: #1242 

/deploy #persist

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -218,6 +218,8 @@ lazy val commonSettings = Seq(
     "-Ycache-macro-class-loader:last-modified", // and macro definitions. This can lead to performance improvements.
     "-Ywarn-value-discard" // Emit warning and location for usages of deprecated APIs.
   ),
+  Compile / console / scalacOptions := (Compile / scalacOptions).value.filterNot(_ == "-Xfatal-warnings"),
```
Comment (jachro) [inline build.sbt:221]: I guess we'd like to take the filter out at the end.

File: entities-search/src/test/scala/io/renku/entities/search/WorkflowsEntitiesFinderSpec.scala [hunk 2]
```diff
@@ -184,4 +185,19 @@ class WorkflowsEntitiesFinderSpec
       wfs.map(_.workflowType) should contain theSameElementsAs List(WorkflowType.Step, WorkflowType.Composite)
     }
   }
+
```
Comment (jachro) [inline entities-search/src/test/scala/io/renku/entities/search/WorkflowsEntitiesFinderSpec.scala]: Shouldn't the new test go to a new `ProjectEntitiesFinderSpec`?

File: build.sbt (hunk 1 above)
Comment (eikek) [reply build.sbt:221]: This only applies when you run `console` - compile and test scalac settings are not changed.

File: entities-search/src/test/scala/io/renku/entities/search/WorkflowsEntitiesFinderSpec.scala (hunk 2 above)
Comment (eikek) [reply entities-search/src/test/scala/io/renku/entities/search/WorkflowsEntitiesFinderSpec.scala]: Oh, right 🤦🏼 - I think I wanted it in `EntityFindersSpec`

**Review by jachro — APPROVED** [review]
The changes look really good! It's been quite a bit chunk of work. Thanks Eike!

File: entities-search/src/test/scala/io/renku/entities/search/WorkflowsEntitiesFinderSpec.scala (hunk 2 above)
Comment (jachro) [reply entities-search/src/test/scala/io/renku/entities/search/WorkflowsEntitiesFinderSpec.scala]: Yes, that's also good.

**Pull request merged**